### PR TITLE
Fix #221: Add guard for importing stale data

### LINKED ISSUE #221: Guard against importing old data on main summary DAG backfills
Filling this issue in response to the situation we've encountered here: https://bugzilla.mozilla.org/show_bug.cgi?id=1392246

Basically, we missed two days of main summary runs this weekend and while the 2017-08-20 run started automatically once the underlying issue was fixed, 2017-08-19 had to be triggered manually afterward, which would result in stale data being imported into the experiment viewer (except I knew this already and marked the 2017-08-19 import task as successful immediately after triggering the DAG.) This is just an example, but it's very possible we'll want to run backfills for Main Summary days in the future on an ad-hoc basis.

Ideally, if the date that's being passed in is less than the submission date from the last import we should exit early (which a success condition/log.) We could also add a command line flag that would force import regardless of the last import date in case we really do want that behavior.

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: notebooks/import.py [hunk 1]
```diff
@@ -42,7 +46,20 @@ def get_database_connection():
...
+    cursor.execute(sql, params)
+    result = cursor.fetchone()
+    if result:
+        prior_import_date = result[0]
+        process_date = datetime.datetime.strptime(date, '%Y%m%d').date()
+        if process_date <= prior_import_date:
+            raise StaleImportError(
+                'Process date %s is older than previous import date %s' % (
+                    process_date, prior_import_date))
+
```
Comment (sunahsuh) [inline notebooks/import.py]: Don't we also need to change what we're passing into the `date` column below to be the `date` var?

Comment (robhudson) [inline notebooks/import.py]: I didn't change that, but yeah, you're right. I'll fix that in a follow up commit.

**Comment by robhudson** [comment]
Is it easy to pass command line switches in via Airflow on backfills?

**Comment by sunahsuh** [comment]
We'd probably want to make a change to the job shell script to allow it, but sure? I was thinking this would be more of an issue with ad-hoc runs in weird situations

**Pull request merged**